src/hooks/safe-apps/useChainFromQueryParams.ts

```diff
@@ -0,0 +1,35 @@
+import { useMemo } from 'react'
+import { useRouter } from 'next/router'
+import { ChainInfo } from '@gnosis.pm/safe-react-gateway-sdk'
+import useChains from '@/hooks/useChains'
+
+type ReturnType = {
+  chain?: ChainInfo
+  validChain: boolean
+  loading: boolean
+  error: string | undefined
+}
+
+const useChainFromQueryParams = (): ReturnType => {
```

Comment:
How is this different from the existing `useChainId`?

Reply:
`useChainId` doesn't guarantee that it will return the chain id from the query param but the user flow is dependent on `chainId` from the query params

Reply:
What does `useCurrentChain` lack from your requirements? It defaults to the last chainId if no query present. Isn't that good for the share page as well?

Reply:
That's open for discussion in the future. With is the current implemented in `safe-react` the app is shared for a specific chain

Reply:
If the app doesn't exist on last chainId, it's not really good